File: learn cpp 2024/pythonIntegration/count500M/count500M.py
import time

start_time = time.time()

total = 0
for i in range(1, 500000001):
    total += 1
end_time = time.time()
duration = end_time - start_time

print(f"Counting to 500 million and incrementing a variable took {duration:.6f} seconds. Final value: {total}")

File: learn cpp 2024/pythonIntegration/count500M/count500M.cpp
#include <iostream>
#include <chrono>

int main() {
    auto start_time = std::chrono::high_resolution_clock::now();

    long long int total = 0;
    for (long long int i = 1; i <= 500000000LL; ++i) {
        total++;
    }

    auto end_time = std::chrono::high_resolution_clock::now();
    auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end_time - start_time);

    std::cout << "Counting to 500 million and incrementing a variable took " << duration.count()
              << " milliseconds. Final value: " << total << std::endl;

    return 0;
}
